frontend/app/controllers/spree/checkout_controller.rb

```diff
@@ -25,33 +25,61 @@ class CheckoutController < Spree::StoreController
 
     # Updates the order and advances to the next state (when possible.)
     def update
-      if OrderUpdateAttributes.new(@order, update_params, request_env: request.headers.env).apply
-        @order.temporary_address = !params[:save_user_address]
-        success = if @order.state == 'confirm'
-          @order.complete
-        else
-          @order.next
-        end
-        if !success
-          flash[:error] = @order.errors.full_messages.join("\n")
-          redirect_to(checkout_state_path(@order.state)) && return
-        end
+      if order_update
+
+        assign_temp_address
+
+        on_failure and return unless state_transition
```

Comment:
You might want to consider renaming `order_update`, `state_transition` and `on_failure` to their verb first equivalents to follow the schematics on the rest of the methods and to indicate that an action is happening. Say to `update_order`, `transition_forward!` and `redirect_on_failure`